pkg/edged/edged.go

```diff
@@ -327,6 +336,8 @@ func (e *edged) StartGarbageCollection() {
 	}, ImageGCPeriod, utilwait.NeverStop)
 
 	go utilwait.Until(func() {
+		e.podReadyLock.RLock()
+		defer e.podReadyLock.RUnlock()
```

Comment:
If we try to acquire the lock in containerGCManager.GarbageCollect() as well, there would be a deadlock problem. Right?